system/Database/DataConverter/Cast/BooleanCast.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\Database\DataConverter\Cast;
+
+/**
+ * Class BooleanCast
+ *
+ * DB column: bool|int(0/1) <--> PHP: bool
+ */
+class BooleanCast extends BaseCast
+{
+    /**
+     * {@inheritDoc}
+     */
+    public static function fromDatabase($value, array $params = []): bool
+    {
+        // For PostgreSQL
+        if ($value === 't') {
+            return true;
+        }
```

Comment:
**Optionally** I would add the ability to make the `DataConverter` aware of the DB driver we use with it. That way, we could have specific cast classes for drivers that need it - such as `BooleanCast` for Postgre.

Reply:
You could create a factory for `DataConverter`, and set specific cast handers.
Or just extend `DataConverter` for a specific DB driver.

Reply:
My proposition wasn't about the ability to prepare a custom Cast class, but providing built-in support for data that are handled differently depending on the database we work with. Like, if I work with data from MySQL, the `t` value shouldn't be treated as `true`.

Anyway, I still may be missing the big picture for this class.

Reply:
I think that this DataConverter and the cast handlers can handle most cases for the DBs we have now.
However, if we really need specific cast handlers for Postgres, for example, I would not be opposed to adding it.

Reply:
I'm not opposed to @michalsn 's suggestion but I think the current solution is fine. If a dev asks to cast a field as a bool and it contains a string "f" then it might be slightly confusing to get `false` but it was a mistake either way.

Reply:
There is nothing wrong with the way we're handling this right now... the only disadvantage is that casting may be a bit too wide sometimes. I saw this as an opportunity to fix this. I may or may not come back to this idea later, though it's nothing crucial.